catkin_tools/verbs/catkin_build/build.py

```diff
@@ -445,9 +445,43 @@ def build_isolated_workspace(
             else:
                 break
 
+    # Populate .catkin file if we're not installing
+    # NOTE: This is done to avoid the Catkin CMake code from doing it,
+    # which isn't parallel-safe. Catkin CMake only modifies this file if
+    # it's package source path isn't found.
+    if not context.install:
+        wide_log("[build] Checking devel manifest...")
+        dot_catkin_file_path = os.path.join(context.devel_space_abs, '.catkin')
+        # If the file exists, get the current paths
+        if os.path.exists(dot_catkin_file_path):
+            dot_catkin_paths = open(dot_catkin_file_path, 'r').read().split(';')
+        else:
+            dot_catkin_paths = []
+
+        # Update the list with the new packages (in topological order)
+        packages_to_be_built_paths = [
+            path
+            for path, pkg in packages_to_be_built
+        ]
+
+        new_dot_catkin_paths = [
+            os.path.join(context.source_space_abs, path)
+            for path, pkg in all_packages
+            if path in dot_catkin_paths or path in packages_to_be_built_paths
+        ]
```

Comment:
I'm not sure I care for this approach. The problem I have with it (unless I'm not interpreting the change correctly) is that packages which have not been built yet are already in the `.catkin` file when the first package is built. This is different than what happens normally (without environment caching) and I'm not sure what the consequences might be. At the very least I think it could lead to subtle bugs which only show up (or are only suppressed) when using catkin_tools, but do not when using something like catkin_make_isolated.